iree/compiler/Codegen/Common/PolynomialApproximationPass.cpp

```diff
@@ -0,0 +1,50 @@
+// Copyright 2022 The IREE Authors
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+#include "iree/compiler/Codegen/PassDetail.h"
+#include "iree/compiler/Codegen/Passes.h"
+#include "mlir/Dialect/Math/Transforms/Approximation.h"
+#include "mlir/Dialect/Math/Transforms/Passes.h"
+#include "mlir/Pass/Pass.h"
+#include "mlir/Transforms/GreedyPatternRewriteDriver.h"
+
+namespace mlir {
+namespace iree_compiler {
+
+/// Command line to use native hardware operations instead of polynomial
+/// approximation.
+static llvm::cl::opt<bool> clNativeMathPrecision(
+    "iree-codegen-gpu-native-math-precision",
```

Comment:
This isn't GPU specific? It's also used for CPU.

Reply:
It only really makes sense for gpu since cpu would not have such native ops but the name may be confusing indeed. Any suggestion on a better name?

Reply:
Okay I mixed this command-line option and the pass. The pass is used for both but this command line option is for GPU. Makes sense to me know. Thanks for the clarification!